File: reacciones/solutions/correct/dibanez.py
n, k = [int(x) for x in input().split(' ')]

s = input()

v = [1]*26

ans = []
for c in s:
    i = ord(c) - ord('a')
    if v[i] > k:
        print("imposible")
        exit(0)
    ans.append(str(v[i]))
    v[i] += 1

print(' '.join(ans))

File: reacciones/solutions/correct/dibanez.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);

    vector<int> v(26, 0);

    int n, k;
    cin >> n >> k;
    string s;
    cin >> s;

    vector<int> ans(n, -1);

    for (int i = 0; i < n; ++i) {
        if (v[s[i] - 'a'] >= k) {
            cout << "imposible\n";
            exit(0);
        }
        ans[i] = ++v[s[i] - 'a'];
    }

    for (int i = 0; i < n; ++i)
        // la última parte del cout solo es necesaria si el juez es muy estricto
        cout << ans[i] << " \n"[i==n-1];

    return 0;
}
